aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -67,6 +78,29 @@ using namespace at::sparse_csr;
 // certain utiliy functions are usable from sparse COO.
 using namespace at::sparse;
 
+namespace {
+
+  template <typename F>
+  inline Tensor get_result_tensor_for_unary_op(F op, const Tensor& input) {
+    auto values = input.values();
+
+    // To handle type promotion for inputs to unary ops,
+    // we first get the result from the underlined op, and use the result
+    // to create a sparse CSR tensor, which is used as the input to the out= variant
+    auto result_values = (*op)(values);
```

Comment:
minor: 
I don't think deferencing the `op` is required.

https://godbolt.org/z/T89nMWWh3